test/scripts/run_test.py

```diff
@@ -47,47 +49,58 @@ def analyse_instrumentation_test_result(result):
     return ok_num
 
 
-def analyse_unit_test_result(result):
-    total_num = 0
-    fail_num = 0
+def analyse_unit_test_result(result, mode, module):
+    xml_file_name = module + '-' + mode.value + '-result-' + TIMESTAMP + '.xml'
+    p = re.compile(".*?XML test result file generated at (.*?\.xml)")
+    dst_xml_path = os.path.join(os.path.split(result)[0], xml_file_name)
     with open(result, 'r') as f:
         for line in f:
-            m_failure = re.match('Tests run:\s+(\d+).*?Failures:\s+(\d+)', line)
-            m_succeed = re.match('OK.*?(\d+) test', line)
-            if m_failure is not None:
-                total_num = int(m_failure.group(1))
-                fail_num = int(m_failure.group(2))
-            if m_succeed is not None:
-                total_num = int(m_succeed.group(1))
-    return total_num, fail_num
-
-
-def exec_am_cmd(am_cmd, device, target_package, result_file, logcat_file):
+            m = re.match(p, line)
+            if m is not None:
+                shutil.copyfile(m.group(1), dst_xml_path)
+    try:
+        xml_doc = minidom.parse(dst_xml_path)
+    except:
+        print "Error occured while reading test result."
+        return 0, 0
+    test_suite = xml_doc.documentElement
+    total_num = int(test_suite.attributes["tests"].value)
+    failed_num = int(test_suite.attributes["failures"].value)
+    error_num = int(test_suite.attributes["errors"].value)
+    success_num = total_num - failed_num - error_num
+    return total_num, success_num
+
+
+def exec_cmd(mode, cmd, device, target_package, result_file, logcat_file, cmd_path):
+    if device is not None:
+        os.environ['ANDROID_SERIAL'] = device
+    if mode == TestMode.UNIT:
+        subprocess.call([HOME_PATH + '/gradlew', '-q', 'clean'], cwd=cmd_path)
```

Comment:
No need to `gradlew clean` for instrumentation test?

Reply:
instrumentation test aolso do this, but not in same method